e2e/fuzz/replay/replay_message_reader.go

```diff
@@ -135,28 +160,59 @@ func (r *replayMessageReader) startChunkReading(messagesChannel chan []*ReplayMe
 	}()
 }
 
-// checks if given node is done with execution of messages
-func (r *replayMessageReader) checkIfDoneWithExecution(validatorId pbft.NodeID, msg *pbft.MessageReq) {
-	//when nodes reach sequence that is higher than the sequence in file,
-	//or if it is in constant round change we know we either reached the end of execution or nodes are stuck and there is a problem
-	if msg.View.Sequence > r.lastSequence ||
-		isSequenceInContinuousRoundChange(msg, validatorId) {
+// processMessage checks if message is a timeout, or if all messages have been read from the queue
+func (r *replayMessageReader) processMessage(validatorId pbft.NodeID, msg *pbft.MessageReq) bool {
```

Comment:
I would still prefer to extract this part of checking whether is it a timeout message or not to the replay_message_notifier (where it was prior to this change) and rename this one to markMessageProcessingDone (or something like that) and it doesn't need to return anything as a result. It would just mark node as done within the map if needed and that's all.
My rationale is that this method should do exactly one thing: mark message processing as done if necessary and that's it.